api/types/server.go

```diff
@@ -397,15 +407,21 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 		return trace.BadParameter(`server SubKind must only be set when Kind is "node"`)
 	}
 
+	// DELETE IN 14.0. (Joerger) PublicAddr deprecated in favor of PublicAddrs
+	if len(s.Spec.PublicAddrs) == 0 && s.Spec.PublicAddr != "" {
+		s.Spec.PublicAddrs = []string{s.Spec.PublicAddr}
+		s.Spec.PublicAddr = ""
```

Comment:
In this case the auth server would temporarily be missing the correct `public_addr` until the next server heartbeat. To avoid this, I've changed this to instead have the Auth Server check both `PublicAddr` and `PublicAddrs` instead of altering the node provided field.